### PR TITLE
Add option to item-directive to hide specific relation-types

### PR DESCRIPTION
**Pull request by SteinHeselmans**
Add option to item-directive to hide specific relation-types in the rendered item definition.

### CODE REVIEW

**Comment by Letme** [comment]
Can you also add a test for your new feature?

Should be something sweet, where no-show would not have element there, while without it it would.  Maybe `noshow` could be better name as `hiderelationship` or `hidelinks` ? What is your use case?

**Comment by SteinHeselmans** [comment]
I'll check if i can add a test for it, and consider another name. Not a fan of 'hiderelationship' as you might want to reuse it for hiding attributes as well in the future.

Use case: I have documentation items for every release, and for every feature that we implement. Those link to eachother to mark which features go into which release. We also have documentation items for the V-model stuff (requirements, design, test, ...). The V-model-items link to the feature-item that impacts (added/altered) the feature implementation. So that way, we have some sort of history in our V-model doc. Now, you can imagine that the feature-items can have a lot of links as they link to everything in the V-model documentation. To make those links easier to interpret, i added an item-matrix for every layer in the V-model: feature to requirement, feature to design, feature to test, ... These tables give more sorted output and is more organized compared to what we see on the item rendering itself. So i want to hide (or notshow ;-) ) the relations to the V-model on the epic, to remove this clutter.

**Comment by Letme** [comment]
If I am looking over the docs then we have a lot of `hide` stuff, so `:hidesource:` ,`hidetarget`, `:hidetitle:` in item-matrixes so maybe better would be `:hidelinks:` ?

Looking deeper, we also do have `:nocaptions:` and `:nooverwrite:` for example (yay, no consistency) so we could have `:nolinks:` or `noattributes`. You do not want to mix attributes and relationships in one flag, so I would keep in mind we want `hideattributes` option if there would be a use case for this.

In traceability we have `sourcetype` so `:hidetype:` (or `:notype:` since items have `no` - dont like this one)?

**Comment by SteinHeselmans** [comment]
I think `hidelinks` is most consistent amongst what you found, and indeed `hideattributes` might be a good addition if we ever need it. The hardest part is done, we found a name ;-)

Still working on a new test...

File: doc/usage.rst [hunk 1]
```diff
@@ -54,13 +54,17 @@ Documentation items can be defined using the *item* directive, specifying:
...
+        :hidelinks: validated_by
 
         According to the Polarion reference, the software **shall** implement my first requirement.
 
 Attributes can be added to the item, using the configured attribute keys in :ref:`traceability_default_config`
 (e.g. *value* in the above example). The content of the attribute is treated as a single string and should
 match_ the regular expression in configuration.
 
+The *hidelinks* is a space-seperated list of relation types. The relations to other items will not be shown for the
+mentioned relation types.
```
Comment (Letme) [inline doc/usage.rst:66]:
```suggestion
mentioned relation types below the item. The links will still be used and displayed in traceability matrixes and other directives.
```

**Comment by JasperCraeghs** [comment]
We never use the term "link" in the name of an option. I am strongly in favor of `hidetype`.

```
$ grep -F 'type:' -r mlx -I
mlx/traceability/directives/item_relink_directive.py:        if not reverse_type:
mlx/traceability/directives/item_relink_directive.py:         :type: relationship_type
mlx/traceability/directives/item_link_directive.py:         :type: relationship_type
mlx/traceability/directives/item_tree_directive.py:         :type: <<relationship>> ...
mlx/traceability/directives/item_2d_matrix_directive.py:         :type: <<relationship>> ...
mlx/traceability/directives/item_pie_chart_directive.py:         :sourcetype: <<relationship>> ...
mlx/traceability/directives/item_pie_chart_directive.py:         :targettype: <<relationship>> ...
mlx/traceability/directives/item_pie_chart_directive.py:         :splitsourcetype:
mlx/traceability/directives/item_matrix_directive.py:         :type: <<relationship>> ...
mlx/traceability/directives/item_matrix_directive.py:         :sourcetype: <<relationship>> ...
```

File: doc/usage.rst [hunk 2]
```diff
@@ -54,13 +54,17 @@ Documentation items can be defined using the *item* directive, specifying:
...
 
         According to the Polarion reference, the software **shall** implement my first requirement.
 
 Attributes can be added to the item, using the configured attribute keys in :ref:`traceability_default_config`
 (e.g. *value* in the above example). The content of the attribute is treated as a single string and should
 match_ the regular expression in configuration.
 
+The *hidelinks* is a space-seperated list of relation types. The relations to other items will not be shown for the
+mentioned relation types below the item. The links will still be used and displayed in traceability matrixes and other directives.
+
```
Comment (JasperCraeghs) [inline doc/usage.rst:67]: Please merge this paragraph with the paragraph on line 80-81 like so:

```
:hidetype: *optional*, *multiple arguments (space-separated)*

    A space-separated list of relation types. By default, the rendered item definition displays all relations to other items
    in a list below the item name (id). The relation types given to this option will be omitted from this list.
    It does not affect how the item is rendered in other directives, e.g. item-matrix.

:nocaptions: *optional*, *flag*

    Hides the caption in the rendered item definition.
```
It is my intention to document all options of all directives in this way. Do you like it?

**Review by Letme — APPROVED** [review]
Love the whole sphinx path test. Thanks for your contribution and for improving our test coverage :wink:

I will let Jasper do another review as he had more comments, but LGTM.

File: doc/usage.rst (hunk 2 above)
Comment (SteinHeselmans) [reply doc/usage.rst]: yep looks good, and uniformity is even better ;-)

File: mlx/traceability/directives/item_directive.py [hunk 3]
```diff
@@ -63,7 +63,8 @@ def _process_relationships(self, *args):
         All targets get naturally sorted per relationship.
         """
         for rel, targets in self._item.all_relations:
-            self._list_targets_for_relation(rel, targets, *args)
+            if rel not in self['hidetype']:
+                self._list_targets_for_relation(rel, targets, *args)
```
Comment (JasperCraeghs) [inline mlx/traceability/directives/item_directive.py:67]: Do you want a warning logged when a relation type in `self['hidetype']` is not in `self._item.all_relations`? 

Keep in mind that it is possible to conditionally change/add relations with the `only`-directive and `item-relink`/`item-link` directive. And you can't really use the `:hidetype:` conditionally.

Comment (SteinHeselmans) [reply mlx/traceability/directives/item_directive.py:67]: checking against `self._item.all_relations()` will not work for my use case, as the relation-type might not be there for this particular item. We just want, if it's there don't show it...
I added a check for the hidetype being present in `TraceableCollection::relations`, which we also do for other places where relation-types are given (item-tree, item-matrix, ...).

**Review by JasperCraeghs — APPROVED** [review]

**Pull request merged**